src/v/debug_bundle/debug_bundle_service.h

```diff
@@ -106,9 +111,90 @@ class service final : public ss::peering_sharded_service<service> {
 
 private:
+    /**
+     * @brief Constructs the arguments for the rpk debug bundle command
+     *
+     * @param job_id Job ID
+     * @param params parameters
+     * @return std::vector<ss::sstring> The list of strings to pass to
+     * external_process
+     */
+    std::vector<ss::sstring>
+    build_rpk_arguments(uuid_t job_id, debug_bundle_parameters params);
+
+    /**
+     * @brief Returns the status of the running process
+     *
+     * @return std::optional<debug_bundle_status> Will return std::nullopt if
+     * process was never executed, else will return the debug_bundle_status
+     */
+    std::optional<debug_bundle_status> process_status() const;
+
+    /**
+     * @return Whether or not the RPK debug bundle process is running
+     */
+    bool is_running() const;
+
+private:
+    /// Handler used to emplace stdout/stderr into a buffer
+    struct output_handler {
+        using consumption_result_type =
+          typename ss::input_stream<char>::consumption_result_type;
+        using stop_consuming_type =
+          typename consumption_result_type::stop_consuming_type;
+        using tmp_buf = stop_consuming_type::tmp_buf;
+
+        std::reference_wrapper<chunked_vector<ss::sstring>> output_buffer;
+
+        ss::future<consumption_result_type> operator()(tmp_buf buf) {
+            output_buffer.get().emplace_back(
+              ss::sstring{buf.begin(), buf.end()});
+            co_return ss::continue_consuming{};
+        }
+    };
+    /// Structure used to hold information about the running rpk debug bundle
+    /// process
+    struct debug_bundle_process {
+        uuid_t job_id;
+        std::unique_ptr<external_process::external_process> rpk_process;
+        std::optional<ss::experimental::process::wait_status> wait_result;
+        std::filesystem::path output_file_path;
+        chunked_vector<ss::sstring> cout;
+        chunked_vector<ss::sstring> cerr;
+        clock::time_point created_time;
+
+        debug_bundle_process(
+          uuid_t job_id,
+          std::unique_ptr<external_process::external_process> rpk_process,
+          std::filesystem::path output_file_path)
+          : job_id(job_id)
+          , rpk_process(std::move(rpk_process))
+          , output_file_path(std::move(output_file_path))
+          , created_time(clock::now()) {}
+
+        debug_bundle_process() = delete;
+        debug_bundle_process(debug_bundle_process&&) = default;
+        debug_bundle_process& operator=(debug_bundle_process&&) = default;
+        debug_bundle_process(const debug_bundle_process&) = delete;
+        debug_bundle_process& operator=(const debug_bundle_process&) = delete;
+
+        ~debug_bundle_process() {
+            if (rpk_process) {
+                vassert(
+                  !rpk_process->is_running(),
+                  "Destroying process struct without waiting for process to "
+                  "finish");
+            }
+        }
+    };
```

Comment:
These can be forward declared, but you need to make some minor changes:
```c++
    ~service() noexcept;
    /// External process
    std::unique_ptr<debug_bundle_process> _rpk_process;
```
And then you can reduce headers and maybe move to some implementation_deps.


`debug_bundle_process` has a constructor, so maybe make it a class and put the members at the end?